ha-sip/src/call.py

```diff
@@ -145,24 +151,46 @@ def handle_events(self) -> None:
             return
         if self.playback_is_done and self.scheduled_post_action:
             post_action = self.scheduled_post_action
-            self.scheduled_post_action = None
-            log(self.account.config.index, 'Scheduled post action: %s' % post_action)
+            if post_action not in ['repeat_message', 'repeat_playlist']:
+                log(self.account.config.index, 'Scheduled post action: %s' % post_action)
             if post_action == 'noop':
                 pass
             elif post_action == 'return':
                 self.handle_menu(self.menu['parent_menu'])
             elif post_action == 'hangup':
                 self.hangup_call()
             elif post_action == 'repeat_message':
-                self.handle_menu(self.menu, send_webhook_event=False, handle_action=False, reset_input=False)
+                pl = self.menu['playlist']
+                if pl:
+                    item_type, content = pl[-1].split(':', 1)
+                    self.menu[item_type] = content.lstrip()
```

Comment:
I think here you're mutating the in-memory menu and if you come back later through some `return` action you don't get the original playlist anymore?